main_test.go

```diff
@@ -0,0 +1,291 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	testConsent "identity_platform_login_ui/ory_mocking/Consent"
+	testErrors "identity_platform_login_ui/ory_mocking/Errors"
+	testLoginUpdate "identity_platform_login_ui/ory_mocking/Login"
+	testLoginBrowser "identity_platform_login_ui/ory_mocking/LoginBrowser"
+	testServers "identity_platform_login_ui/ory_mocking/Testservers"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+// --------------------------------------------
+// TESTING WITH CORRECT SERVERS
+// --------------------------------------------
+func TestHandleCreateFlowWithoutCookie(t *testing.T) {
+	//init clients
+	serverClose := testServers.CreateTestServers()
+	defer serverClose()
+
+	//create request and response objects
+	req := httptest.NewRequest(http.MethodGet, "/api/kratos/self-service/login/browser?aal=aal1&login_challenge=&refresh=false&return_to=http://test.test", nil)
+	req.Header.Set("Content-Type", "application/json")
+	w := httptest.NewRecorder()
+
+	//test and evaluate test
+	handleCreateFlow(w, req)
+	res := w.Result()
+	defer res.Body.Close()
+	data, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		t.Errorf("expected error to be nil got %v", err)
+	}
+	lbr := new(testLoginBrowser.LoginBrowserResponse)
+	if err := json.Unmarshal(data, lbr); err != nil {
+		t.Errorf("expected error to be nil got %v", err)
+	}
+	if lbr.Id != "test_id" {
+		t.Errorf("expected test_id, got %v", string(data))
+	}
+}
+
+func TestHandleCreateFlowWithCookie(t *testing.T) {
+	//init clients
+	serverClose := testServers.CreateTestServers()
+	defer serverClose()
+
+	//create request and response objects
+	req := httptest.NewRequest(http.MethodPut, "/api/kratos/self-service/login/browser?aal=aal1&login_challenge=test_challange&refresh=false&return_to=http://test.test", nil)
+	req.Header.Set("Content-Type", "application/json")
+	cookie := &http.Cookie{
+		Name:   "ory_kratos_session",
+		Value:  "test-token",
+		MaxAge: 300,
+	}
+	req.AddCookie(cookie)
+	w := httptest.NewRecorder()
+
+	//test and evaluate test
+	handleCreateFlow(w, req)
+	res := w.Result()
+	defer res.Body.Close()
+	data, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		t.Errorf("expected error to be nil got %v", err)
+	}
+	if res.StatusCode != 200 {
+		t.Errorf("expected StatusCode to be 200 got %v", res.StatusCode)
+	}
+	requestLoginResponse := new(testLoginBrowser.OAuth2RequestLoginResponse)
+	if err := json.Unmarshal(data, requestLoginResponse); err != nil {
+		t.Errorf("expected error to be nil got %v", err)
+	}
+	if requestLoginResponse.Redirect_to != "test.test" {
+		t.Errorf("expected test.test, got %v", string(data))
+	}
```

Comment:
You should use [assertions](https://pkg.go.dev/github.com/stretchr/testify/assert) to verify the response's output

Reply:
Changed if's to assertions.